parcels/kernel.py

```diff
@@ -239,9 +239,10 @@ def __init__(
 
         numkernelargs = self.check_kernel_signature_on_version()
 
-        assert (
-            numkernelargs == 3
-        ), "Since Parcels v2.0, kernels do only take 3 arguments: particle, fieldset, time !! AND !! Argument order in field interpolation is time, depth, lat, lon."
+        if numkernelargs != 3:
+            raise ValueError(
+                "Since Parcels v2.0, kernels do only take 3 arguments: particle, fieldset, time !! AND !! Argument order in field interpolation is time, depth, lat, lon."
```

Comment:
I thought we wanted to keep this `ValueError`, because otherwise running old Kernels can really get users into trouble...

Reply:
Just changed it from assertion to value error. TODO: Add unit test so we avoid regression